presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -83,8 +84,11 @@ private DateTimeFunctions() {}
     @SqlType(StandardTypes.DATE)
     public static long currentDate(ConnectorSession session)
     {
-        long millis = getChronology(session.getTimeZoneKey()).dayOfMonth().roundFloor(session.getStartTime());
-        return MILLISECONDS.toDays(millis);
+        ISOChronology chronology = getChronology(session.getTimeZoneKey());
+
+        DateTime currentDateTime = new DateTime(System.currentTimeMillis(), chronology).withTimeAtStartOfDay();
+        DateTime baseDateTime = new DateTime(1970, 1, 1, 0, 0, chronology).withTimeAtStartOfDay();
+        return Days.daysBetween(baseDateTime, currentDateTime).getDays();
```

Comment:
I think, just adding the current time offset to the millisecond of local start date - will work also. 
Will look tomorrow are there any corner cases.